raiden/tasks.py

```diff
@@ -1,17 +1,33 @@
 import time
-import structlog
+import threading
+from pkg_resources import parse_version, require
 
 import gevent
 from gevent.event import AsyncResult
-from gevent.queue import (
-    Queue,
-)
+from gevent.queue import Queue
+import structlog
+
 from raiden.exceptions import RaidenShuttingDown
 
+CHECK_VERSION_INTERVAL = 60 * 5
 REMOVE_CALLBACK = object()
 log = structlog.get_logger(__name__)  # pylint: disable=invalid-name
 
 
+def check_version():
+    """Check every 3 hours for a new release"""
+    app_version = parse_version(require('raiden')[0].version)
+    while True:
+        content = grequests.get('https://api.github.com/repos/raiden-network/raiden/releases/latest').json()
```

Comment:
We need error handling here (request errors and json decode) otherwise this thread can crash the whole application if github briefly isn't available (or the user has a brief networking disconnect)